src/stitching/mergeSchemas.ts

```diff
@@ -0,0 +1,653 @@
+    },
+  });
+  typeRegistry.setQuery(query);
+  let mutation;
+  if (!isEmpty(mutationFields)) {
+    mutation = new GraphQLObjectType({
+      name: 'Mutation',
+      fields: {
+        ...mutationFields,
+      },
+    });
+    typeRegistry.setMutation(mutation);
+  }
+
+  return new GraphQLSchema({
+    query,
+    mutation,
+  });
+}
+
+function prefixName(name: string, prefix: string) {
+  return `${prefix}_${name}`;
+}
+
+function recreateCompositeType(
+  schema: GraphQLSchema,
+  type: GraphQLCompositeType,
+  registry: TypeRegistry,
+): GraphQLCompositeType {
+  if (type instanceof GraphQLObjectType) {
+    const fields = type.getFields();
+    // XXX we don't really support interfaces yet
+    const interfaces = type.getInterfaces();
+    let interfaceNames: Array<string>;
+    if (interfaces) {
+      interfaceNames = interfaces.map(iface => iface.name);
+    } else {
+      interfaceNames = [];
+    }
+    const newType = new GraphQLObjectType({
+      name: type.name,
+      description: type.description,
+      isTypeOf: type.isTypeOf,
+      fields: () => ({
+        ...fieldMapToFieldConfigMap(fields, registry),
+        ...createLinks(registry.getLinksByType(type.name), registry),
+      }),
+      interfaces: () =>
+        interfaceNames.map(() => {
+          throw new Error('Not supporting interfaces yet');
+        }),
+    });
+    return newType;
+  } else {
+    console.warn('We do not support interfaces and union yet.');
+    return type;
+  }
+}
+
+function fieldMapToFieldConfigMap(
+  fields: GraphQLFieldMap<any, any>,
+  registry: TypeRegistry,
+): GraphQLFieldConfigMap<any, any> {
+  return mapValues(fields, field => fieldToFieldConfig(field, registry));
+}
+
+function fieldToFieldConfig(
+  field: GraphQLField<any, any>,
+  registry: TypeRegistry,
+): GraphQLFieldConfig<any, any> {
+  return {
+    type: registry.resolveType(field.type),
+    args: argsToFieldConfigArgumentMap(field.args, registry),
+    description: field.description,
+    deprecationReason: field.deprecationReason,
+  };
+}
+
+function argsToFieldConfigArgumentMap(
+  args: Array<GraphQLArgument>,
+  registry: TypeRegistry,
+): GraphQLFieldConfigArgumentMap {
+  return fromPairs(args.map(arg => argumentToArgumentConfig(arg, registry)));
+}
+
+function argumentToArgumentConfig(
+  argument: GraphQLArgument,
+  registry: TypeRegistry,
+): [string, GraphQLArgumentConfig] {
+  return [
+    argument.name,
+    {
+      type: registry.resolveType(argument.type),
+      defaultValue: argument.defaultValue,
+      description: argument.description,
+    },
+  ];
+}
+
+function createLinks(
+  links: Array<SchemaLink>,
+  registry: TypeRegistry,
+): { [key: string]: GraphQLFieldConfig<any, any> } {
+  if (!registry.query) {
+    throw new Error('Somehow we do not have Query in registry');
+  }
+  const queryFields = registry.query.getFields();
+  return fromPairs(
+    links.map(link => {
+      const [schemaName, field] = link.to.split('_');
```

Comment:
Yeah, whole namespacing related code is a bit of a hack and will be better once we figure out proper namespaces.